### ISSUE TITLE
warning-as-error should prevent execution

### ISSUE THREAD

**Issue by jaredly** [ScalaConsultants/replumb#121]
Currently, the code is executed even if there are warnings, and only if there's no error are warnings reported. This leads to more confusing errors in some cases.

For example:
`(no-a-thing 10)` results in the error `ERROR` w/ underlying error `cannot read property call of undefined`, whereas the warning that cljs generates is much more helpful `use of undefined var "cljs.user/not-a-thing"`.

Possible solutions:
- switch to a two-phase "analyze" then "eval" system
- if warnings are provided in the [analysis cache](https://github.com/clojure/clojurescript/blob/master/src/main/cljs/cljs/js.cljs#L86) that `*eval-fn*` receives, then eval-fn could just abort when warnings are found

**Comment by arichiardi** [comment]
Right! Probably a custom `\*eval-fn\*` (so second option) is the way I would go. In any case we will need it for implementing the eval cache (which is the last big feature missing) in replumb vs planck
